addons/mail/models/mail_link_preview.py

```diff
@@ -89,37 +89,44 @@ def _is_domain_throttled(self, url):
     @api.model
     def _get_link_preview_from_url(self, url, request_session):
         try:
-            response = request_session.head(url, timeout=3, allow_redirects=True)
+            # Some websites are blocking non browser user agent.
+            request_session.headers.update({'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'})
+            response = request_session.get(url, timeout=3, allow_redirects=True, stream=True)
         except requests.exceptions.RequestException:
             return False
-        if response.status_code != requests.codes.ok:
+        if not response.ok or not response.headers.get('Content-Type'):
             return False
-        image_mimetype = (
-            'image/bmp',
-            'image/gif',
-            'image/jpeg',
-            'image/png',
-            'image/tiff',
-            'image/x-icon',
-        )
         # Content-Type header can return a charset, but we just need the
         # mimetype (eg: image/jpeg;charset=ISO-8859-1)
         content_type = response.headers['Content-Type'].split(';')
-        if response.headers['Content-Type'].startswith(image_mimetype):
+        if response.headers['Content-Type'].startswith('image/'):
             return {
                 'image_mimetype': content_type[0],
+                'og_image': url, # If the url mimetype is already an image type, set url as preview image
                 'source_url': url,
             }
-        if response.headers['Content-Type'].startswith('text/html'):
-            return self._get_link_preview_from_html(url, request_session)
+        elif response.headers['Content-Type'].startswith('text/html'):
+            return self._get_link_preview_from_html(url, response)
         return False
 
-    def _get_link_preview_from_html(self, url, request_session):
-        response = request_session.get(url, timeout=3)
-        parser = etree.HTMLParser(encoding=response.encoding)
-        tree = html.fromstring(response.content, parser=parser)
+    def _get_link_preview_from_html(self, url, response):
+        content = b""
+        for chunk in response.iter_content(chunk_size=8192):
```

Comment:
does it work ? (so you don't need to decode below)
```suggestion
        for chunk in response.iter_content(chunk_size=8192, decode_unicode=True):
```